include/deal.II/distributed/cell_weights.h

```diff
@@ -38,80 +38,178 @@ namespace parallel
    * This class allows computing these weights for load balancing by
    * consulting the FiniteElement that is associated with each cell of
    * a hp::DoFHandler. One can choose from predefined weighting
-   * algorithms provided by this class or provide a custom one. The
-   * chosen weighting function will be connected to the corresponding
-   * signal of the linked parallel::TriangulationBase via callback.
+   * algorithms provided by this class or provide a custom one.
    *
-   * An object of this class needs to exist for every DoFHandler associated
-   * with the Triangulation we work on to achieve satisfying work balancing
-   * results.
-   *
-   * A small code snippet follows explaining how to achieve each cell
-   * being weighted by its current number of degrees of freedom.
+   * This class offers two different ways of connecting the chosen weighting
+   * function to the corresponding signal of the linked
+   * parallel::TriangulationBase. The recommended way involves creating an
+   * object of this class which will automatically take care of registering the
+   * weighting function upon creation and de-registering it once destroyed. An
+   * object of this class needs to exist for every DoFHandler associated with
+   * the Triangulation we work on to achieve satisfying work balancing results.
+   * The connected weighting function may be changed anytime using the
+   * CellWeights::reinit() function. The following code snippet demonstrates how
+   * to achieve each cell being weighted by its current number of degrees of
+   * freedom. We chose a factor of `1000` that corresponds to the initial weight
+   * each cell is assigned to upon creation.
    * @code
-   * parallel::CellWeights<dim, spacedim> cell_weights(hp_dof_handler);
-   * cell_weights.register_ndofs_weighting();
+   * parallel::CellWeights<dim, spacedim> cell_weights(
```

Comment:
could it be `const`?

Reply:
Unfortunately not.

`register_ndofs_weighting()` is a non-const function and changes class members, i.e. `connection`.